src/overlays/actors/ovl_En_Anubice/z_en_anubice.c

```diff
@@ -53,35 +63,441 @@ static ColliderCylinderInit D_809B22D0 = {
+    f32 lastFrame = Animation_GetLastFrame(&D_06000348);
+
+    this->animLastFrame = lastFrame;
+    Animation_Change(&this->skelAnime, &D_06000348, 1.0f, 0.0f, lastFrame, 2, -20.0f);
+
+    this->unk_256 = 0;
+    this->unk_258 = 0;
+    this->deathTimer = 20;
+    this->actor.velocity.z = 0.0f;
+    this->actor.velocity.x = 0.0f;
+    this->actor.gravity = -1.0f;
+
+    if (BgCheck_SphVsFirstPoly(&globalCtx->colCtx, &this->fireballPos, 70.0f)) {
+        this->unk_256 = 1;
+        this->unk_258 = this->actor.shape.rot.x - 0x7F00;
+    }
+
+    this->actionFunc = &EnAnubice_Die;
+}
+
+void EnAnubice_Die(EnAnubice* this, GlobalContext* globalCtx) {
+    f32 curFrame;
+    f32 phi_f2;
+    Vec3f sp4C;
+    Vec3f fireEffectPos;
+    s32 pad;
+
+    sp4C = D_809B231C;
+    fireEffectPos = D_809B2328;
+
+    SkelAnime_Update(&this->skelAnime);
+    Math_ApproachZeroF(&this->actor.shape.shadowScale, 0.4f, 0.25f);
+
+    if (this->unk_256) {
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->unk_258, 1, 10000, 0);
+        if (fabsf(this->actor.shape.rot.y - this->unk_258) < 100.0f) {
+            this->unk_256 = 0;
+        }
+    }
+
+    curFrame = this->skelAnime.curFrame;
+    phi_f2 = curFrame * -3000.0f;
+    phi_f2 = MAX(-11000.0f, phi_f2);
+
+    Matrix_RotateY((this->actor.shape.rot.y / 32768.0f) * M_PI, MTXMODE_NEW);
+    Matrix_RotateX((phi_f2 / 32768.0f) * M_PI, MTXMODE_APPLY);
+    sp4C.y = Rand_CenteredFloat(10.0f) + 30.0f;
+    Matrix_MultVec3f(&sp4C, &fireEffectPos);
+    fireEffectPos.x += this->actor.world.pos.x + Rand_CenteredFloat(40.0f);
+    fireEffectPos.y += this->actor.world.pos.y + Rand_CenteredFloat(40.0f);
+    fireEffectPos.z += this->actor.world.pos.z + Rand_CenteredFloat(30.0f);
+    func_8003426C(&this->actor, 0x4000, 0x80, 0, 8);
+    EffectSsEnFire_SpawnVec3f(globalCtx, &this->actor, &fireEffectPos, 100, 0, 0, -1);
+
+    if (this->animLastFrame <= curFrame && (this->actor.bgCheckFlags & 1)) {
+        Math_ApproachF(&this->actor.shape.yOffset, -4230.0f, 0.5f, 300.0f);
+        if (this->actor.shape.yOffset < -2000.0f) {
+            Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.world.pos, 0xC0);
+            Actor_Kill(&this->actor);
+        }
+    }
+}
+
+void EnAnubice_Update(Actor* thisx, GlobalContext* globalCtx) {
+    f32 zero;
+    BgHidanCurtain* hidanCurtain;
+    s32 i;
+    Vec3f sp48;
+    Vec3f sp3C;
+    EnAnubice* this = THIS;
+
+    if (this->actionFunc != EnAnubice_SetupDie && this->actionFunc != EnAnubice_Die &&
+        this->actor.shape.yOffset == 0.0f) {
+        EnAnubice_Hover(this, globalCtx);
+        for (i = 0; i < 5; i++) {
+            hidanCurtain = this->hidanCurtains[i];
+
+            if (hidanCurtain != NULL && fabsf(hidanCurtain->actor.world.pos.x - this->actor.world.pos.x) < 60.0f &&
+                fabsf(this->hidanCurtains[i]->actor.world.pos.z - this->actor.world.pos.z) < 60.0f &&
```

Comment:
shouldn't this be `hidanCurtain->actor.world.pos.z`?

Reply:
This is required for regalloc.

Reply:
that suggests the temp might not exist at all then, make sure to double check that just incase

Reply:
No temp has the same regalloc as everywhere using the temp.